lib/util/validationResponse.ts

```diff
@@ -3,10 +3,28 @@
 
 import * as pointer from "json-pointer"
 import { Error } from "./error"
+import { Unknown } from "./unknown"
+import { Map } from "./utils"
+
+interface ValidationError {
+  validationCategory: string
+  code?: string
+  providerNamespace: Unknown
+  type: string
+  inner?: Error|Error[]
+  id?: Unknown
+  message?: string
+  jsonref?: string
+  "json-path"?: string
+}
+
+interface Warning {
+  readonly code: Unknown
+}
 
 export class ValidateResponse {
 
-  private mapper = {
+  private readonly mapper: Map<string> = {
     SWAGGER_SCHEMA_VALIDATION_ERROR: "M6000",
```

Comment:
>SWAGGER_SCHEMA_VALIDATION_ERROR [](start = 4, length = 31)

I was wondering if we could have an enum for all the errors

Reply:
we could put the codes as the the enum values `SWAGGER_SCHEMA_VALIDATION_ERROR= "M6000`

---
In reply to: [194478201](https://github.com/Azure/oav/pull/257#discussion_r194478201) [](ancestors = 194478201)